scripts/api/ingest.ts

```diff
@@ -0,0 +1,61 @@
+import {flatMap} from 'lodash';
+import {IIngestRuleHandler, IExtensionActivationResult} from 'superdesk-api';
+import {extensions} from 'appConfig';
+import {gettext} from 'core/utils';
+
+const DEFAULT_HANDLER: IIngestRuleHandler = {
+    name: 'desk_fetch_publish',
+    label: gettext('Desks'),
+    supportedActions: {
+        fetch_to_desk: true,
+        publish_from_desk: true,
+    },
+    supportedConfigs: {
+        exit: true,
+        preserveDesk: true,
+    },
+    getDefaults: function() {
+        return {
+            name: '',
+            handler: 'desk_fetch_publish',
+            filter: null,
+            actions: {
+                fetch: [],
+                publish: [],
+                exit: false,
+            },
+            schedule: {
+                day_of_week: ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN'],
+                hour_of_day_from: null,
+                hour_of_day_to: null,
+                _allDay: true,
+            },
+        };
+    },
+};
+
+function getIngestRuleHandlers() {
```

Comment:
I would rather fetch those from the server

Reply:
Good point, I'll do add the endpoint tomorrow. Though I will still need part of this code, to allow extensions to register the UI that handles the specific ingest rule. Otherwise we'll need the specific UI's for each ingest rule in core